File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.java
class Solution {
    public int maxSum(int[] nums) {
        int[] count = new int[101];
        int max = Integer.MIN_VALUE;
        int sum = 0;
        for(int num:nums){
          if(num>0){
            if(count[num]==0){
              sum +=num;
            }
            count[num]++;
          }
          max = Math.max(max,num);
        }
        return max<0?max:sum;
    }
}

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.py
class Solution(object):
    def maxSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_ = max(nums)
        if max_ < 0:
          return max_
        seen = set(num for num in nums if num>0)
        return sum(seen)
        